### ISSUE TITLE
Idea: Speech Bubble Zoom

### ISSUE THREAD

**Issue by ImAmirx** [KotatsuApp/Kotatsu#1143]
### Describe your suggested feature

Would it be possible to add a speech bubble detector/zoomer to Kotatsu like the one in the Seeneva? 
https://github.com/Seeneva/seeneva-reader-android?tab=readme-ov-file#speech-balloons-zooming

I think it'd be a cool feature

### Acknowledgements

- [X] This is not a duplicate of an existing issue. Please look through the list of [open issues](https://github.com/KotatsuApp/Kotatsu/issues) before creating a new one.

**Comment by Mhuzaifakhalid** [comment]
ok this looks dope

**Comment by MariusAlbrecht** [comment]
do you have any experience with how long this takes? Kotatsu doesn't manage a library itself but loads media on demand.
That means the user would have to wait for the model to do its analysis _while reading_ because it can't really be precomputed like in your example.

**Comment by ImAmirx** [comment]
> do you have any experience with how long this takes? Kotatsu doesn't manage a library itself but loads media on demand. That means the user would have to wait for the model to do its analysis _while reading_ because it can't really be precomputed like in your example.

Would it work if it was only available for the chapters you downloaded and selected to be scanned?

**Comment by galpt** [comment]
AI/ML is compute-heavy. This will drain your battery very quickly for just zooming the speech balloons.
You want to trade longer battery life for such feature? I prefer longer battery life, and not all the time people have access to easily charge their phones. I'm quite against this feature.

It could reduce your reading hours from let's say 4 hours to just 1-2 hours because your GPU is constantly used for _analyzing_ the images. That's indeed heavier than _just displaying_ the images without analyzing them.

It's a cool feature, but reducing my reading hours is what makes me hesitate if it's a wise feature for a phone

**Comment by ImAmirx** [comment]
> AI/ML is compute-heavy. This will drain your battery very quickly for just zooming the speech balloons. You want to trade longer battery life for such feature? I prefer longer battery life, and not all the time people have access to easily charge their phones. I'm quite against this feature.
> 
> It could reduce your reading hours from let's say 4 hours to just 1-2 hours because your GPU is constantly used for _analyzing_ the images. That's indeed heavier than _just displaying_ the images without analyzing them.
> 
> It's a cool feature, but reducing my reading hours is what makes me hesitate if it's a wise feature for a mobile phone

Well, it can be there as a manually activated option for those who want to use it, that way no one loses anything from it being added🤷‍♂️

**Comment by galpt** [comment]
> > AI/ML is compute-heavy. This will drain your battery very quickly for just zooming the speech balloons. You want to trade longer battery life for such feature? I prefer longer battery life, and not all the time people have access to easily charge their phones. I'm quite against this feature.
> > It could reduce your reading hours from let's say 4 hours to just 1-2 hours because your GPU is constantly used for _analyzing_ the images. That's indeed heavier than _just displaying_ the images without analyzing them.
> > It's a cool feature, but reducing my reading hours is what makes me hesitate if it's a wise feature for a mobile phone
> 
> Well, it can be there as a manually activated option for those who want to use it, that way no one loses anything from it being added🤷‍♂️

I agree, this is a nice feature suggestion indeed, I hope the dev will consider it if it's not a hard one to implement
